src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Entry[INITIAL_CAPACITY];
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
+
+        int index = getIndex(key);
+        Entry<K, V> newEntry = new Entry<>(key, value);
+
+        if (table[index] == null) {
+            table[index] = newEntry;
+        } else {
+            Entry<K, V> current = table[index];
+            while (current != null) {
 
+                if (isNullValue(current,key) || isValue(current,key)) {
+                    current.value = value;
+                    return;
+                }
+
+                if (current.next == null) {
+                    current.next = newEntry;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Entry<K, V> current = table[index];
+
+        while (current != null) {
+            if (isValue(current,key) || isNullValue(current,key)) {
```

Comment:
use space after comma
not clear naming, should be smth like "are values equal"
use Objects.equals() instead, or write this code on your own